### PR TITLE
Remove hardcoding in DSP with fileinfo.

### PR DESCRIPTION
**Pull request by calina-c**
Closees #670

### CODE REVIEW

File: ocean_lib/data_provider/data_service_provider.py [hunk 1]
```diff
@@ -180,10 +180,11 @@ def download(
         index: Optional[int] = None,
         userdata: Optional[Dict] = None,
     ) -> None:
-
+        _, fileinfo_endpoint = DataServiceProvider.build_fileinfo(service_endpoint)
```
Comment (mariacarmina) [inline ocean_lib/data_provider/data_service_provider.py:183]: The service_endpoint is the download one. When we call build_fileinfo, we need to pass the root uri of the provider. I suggest to use `get_root_uri` which retruns the provider uri of the service_endpoint => http://172.15.0.4:8030/ and after use the result to call `build_fileinfo`. Maybe #677 can be useful here.

Comment (mariacarmina) [reply ocean_lib/data_provider/data_service_provider.py:183]: This is from asset_downloader file
image

Comment (calina-c) [reply ocean_lib/data_provider/data_service_provider.py:183]: actually, this does open a can of worms: we don't need to be passing in the service endpoint. We can retrieve it from the service inside the download function. We NEVER need provider root uri. We can always infer the root and all related service_endpoints based on the service endpoint. This is applicable to all the functions inside the DSP.

Do you want to take this task or should I work on it tomorrow? I know you are low on tasks.

Comment (calina-c) [reply ocean_lib/data_provider/data_service_provider.py:183]: Look at the snippet you paste it. We are calculating the download endpoint OUTSIDE the download function, then feeding it to the download function as a parameter. But the download function "knows" it's a download function and that it should build the download endpoint. So move that responsibility there. Please get rid of all service_endpoint and provider_root_uri parameters being sent to functions in the DataServiceProvider. We have the service, we can get that on the fly.

**Comment by calina-c** [comment]
@Maria I've created a different ticket for the follow-up: https://github.com/oceanprotocol/ocean.py/issues/680

**Pull request merged**